Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/ReflectometryILLPreprocess.py

```diff
@@ -244,24 +244,31 @@ def validateInputs(self):
             if self.getProperty(Prop.LOW_BKG_WIDTH).value == 0 and self.getProperty(Prop.HIGH_BKG_WIDTH).value == 0:
                 issues[Prop.BKG_METHOD] = 'Cannot calculate flat background if both upper and lower background /' \
                                           ' widths are zero.'
-            if not self.getProperty(Prop.INPUT_WS).isDefault:
-                ws = self.getProperty(Prop.INPUT_WS).value
-                if not ws.run().hasProperty(common.SampleLogs.LINE_POSITION) and \
-                        self.getProperty(Prop.LINE_POSITION_INPUT).isDefault:
-                    issues[Prop.LINE_POSITION_INPUT] = 'Cannot subtract flat background without knowledge of /'\
-                                                       'line position.'
-                    issues[Prop.INPUT_WS] = 'Consider to give a sample log entry ' + common.SampleLogs.LINE_POSITION
         # We cannot use a FloatBoundedValidator here since we need to check for the default empty double
         if not self.getProperty(Prop.LINE_POSITION_INPUT).isDefault:
             beamCentre = self.getProperty(Prop.LINE_POSITION_INPUT).value
             if beamCentre < 0. or beamCentre > 255.:
                 issues[Prop.LINE_POSITION_INPUT] = 'Value should be between 0.0 and 255.0'
+        # Check whether user provides TwoTheta or sample log entry exist
+        if not self.getProperty(Prop.INPUT_WS).isDefault:
+            ws = self.getProperty(Prop.INPUT_WS).value
+            if not ws.run().hasProperty(common.SampleLogs.TWO_THETA):
+                if self.getProperty(Prop.TWO_THETA).isDefault:
+                    issues[Prop.TWO_THETA] = 'Must provide TwoTheta or in sample logs ' + common.SampleLogs.TWO_THETA
+                else:
+                    # Write two theta to sample logs
+                    ws.run().addProperty(common.SampleLogs.TWO_THETA, float(self.getProperty(Prop.TWO_THETA).value),
+                                         'degree', True)
```

Comment:
This method validates the inputs - I don't think it should write any stuff to the sample logs. It is not its responsibility.

The correct place to do this would be in the `_inputWS()` method, I think, in the branch where we read the input workspace property.

Reply:
Yes, I think the same. The input validation should not do anything else.